lib/pbench/test/unit/agent/task/test_results_push.py

```diff
@@ -144,38 +144,34 @@ def test_token_envar(monkeypatch, caplog, valid_config, pytestconfig):
         TestResultsPush.add_http_mock_response()
         caplog.set_level(logging.DEBUG)
         runner = CliRunner(mix_stderr=False)
-        result = runner.invoke(
-            results.results_push, args=[TestResultsPush.CTRL_TEXT, tarball],
-        )
+        result = runner.invoke(main, args=[TestResultsPush.CTRL_TEXT, tarball],)
         assert result.exit_code == 0, result.stderr
-        assert result.stderr_bytes == b"File uploaded successfully\n"
+        assert result.stderr == "File uploaded successfully\n"
 
     @staticmethod
     @responses.activate
     def test_connection_error(monkeypatch, caplog, valid_config, pytestconfig):
-        """Test normal operation with the token in an environment variable"""
+        """Test handling of connection errors"""
 
         monkeypatch.setenv("PBENCH_ACCESS_TOKEN", TestResultsPush.TOKN_TEXT)
         TestResultsPush.add_connectionerr_mock_response()
         caplog.set_level(logging.DEBUG)
         runner = CliRunner(mix_stderr=False)
-        result = runner.invoke(
-            results.results_push, args=[TestResultsPush.CTRL_TEXT, tarball],
-        )
+        result = runner.invoke(main, args=[TestResultsPush.CTRL_TEXT, tarball],)
         assert result.exit_code == 1
         assert str(result.stderr).startswith("Cannot connect to")
 
     @staticmethod
     @responses.activate
     def test_http_error(monkeypatch, caplog, valid_config, pytestconfig):
-        """Test normal operation with the token in an environment variable"""
+        """Test handling of 404 errors"""
 
         monkeypatch.setenv("PBENCH_ACCESS_TOKEN", TestResultsPush.TOKN_TEXT)
         TestResultsPush.add_http_mock_response(HTTPStatus.NOT_FOUND)
         caplog.set_level(logging.DEBUG)
         runner = CliRunner(mix_stderr=False)
-        result = runner.invoke(
-            results.results_push, args=[TestResultsPush.CTRL_TEXT, tarball],
-        )
+        result = runner.invoke(main, args=[TestResultsPush.CTRL_TEXT, tarball])
         assert result.exit_code == 1
-        assert str(result.stderr).find("Not Found") > -1
+        assert (
+            str(result.stderr).find("Not Found") > -1
+        ), f"stderr: {str(result.stderr)!r}; stdout: {str(result.stdout)!r}"
```

Comment:
`result.stderr` and `result.stdout` _should_ already be strings....

Reply:
Fixed.